libuuu/cmd.cpp

```diff
@@ -415,6 +425,73 @@ int CmdDelay::run(CmdCtx *)
 	return 0;
 }
 
+int CmdConfig::parser(char * /*p*/)
+{
+	try 
+	{
+	    size_t pos = 0;
+	    string param = get_next_param(m_cmd, pos);
+
+	    if (param.find(':') != string::npos)
+		    param = get_next_param(m_cmd, pos);
+
+	    if (str_to_upper(param) != "CONFIG")
+		{
+		    string err = "CmdConfig::parser Uknown Commnd:";
+		    err += param;
+		    set_last_err_string(err);
+		    return -1;
+		}
+
+	    string args = get_next_param(m_cmd, pos);
+	    if (str_to_upper(args) == "USB::HID::MAX_PACKET_SIZE")
+		{
+		    string size = get_next_param(m_cmd, pos);
+		    Conf::GetConf().m_USB.m_HIDMaxTransfer = str_to_uint(size);
+		    cout << "changed USB::HID::MAX_PACKET_SIZE to " << Conf::GetConf().m_USB.m_HIDMaxTransfer << endl;
+		}
+	    else if (str_to_upper(args) == "USB::HID::TIMEOUT" )
+		{
+		    string ms = get_next_param(m_cmd, pos);
+		    Conf::GetConf().m_USB.m_HIDTimeout = str_to_uint(ms);
+		    cout << "changed USB::HID::TIMEOUT to " << Conf::GetConf().m_USB.m_HIDTimeout << endl;
+		}
+	    else if (str_to_upper(args) == "USB::BULK::MAX_PACKET_SIZE" )
+		{
+		    string size = get_next_param(m_cmd, pos);
+		    Conf::GetConf().m_USB.m_BulkMaxTransfer = str_to_uint(size);
+		    cout << "changed USB::BULK::MAX_PACKET_SIZE to " << Conf::GetConf().m_USB.m_BulkMaxTransfer << endl;
+		}
+	    else if (str_to_upper(args) == "USB::BULK::TIMEOUT")
+		{
+		    string ms = get_next_param(m_cmd, pos);
+		    Conf::GetConf().m_USB.m_BulkTimeout = str_to_uint(ms);
+		    cout << "changed CONF::USB::BULK::TIMEOUT to " << Conf::GetConf().m_USB.m_BulkTimeout << endl;
+		}
+	    else
+		{
+		    string err = "CmdConfig::parser invalid parameter:";
+		    err += m_cmd;
+		    set_last_err_string(err);
+		    return -1;
+
+		}
```

Comment:
Dose CmdBase::insert_param_info match your parser requirement?  That simplify parser work.